openproblems/tasks/spatial_decomposition/api.py

```diff
@@ -0,0 +1,75 @@
+from ...data.sample import load_sample_data
+from pandas.core.dtypes.common import is_categorical_dtype
+
+import numpy as np
+import pandas as pd
+
+EPS = 1e-10
+
+
+def check_dataset(adata):
+    """Check that dataset output fits expected API."""
+    # test for spatial coordinates
+    assert "spatial" in adata.obsm
+    # check that proportions are included
+    assert "proportions_true" in adata.obsm
+    # make sure proportions sum to one, some precision error allowed
+    assert np.all(
+        np.abs(
+            np.sum(
+                adata[adata.obs["modality"] == "sp"].obsm["proportions_true"], axis=1
+            )
+            - 1  # noqa: W503
+        )
+        < EPS  # noqa: W503
+    )
+    # ensure cell type labels are found in single cell reference
+
+    assert is_categorical_dtype(adata.obs["label"])
+    return True
+
+
+def check_method(adata):
+    """Check that method output fits expected API."""
+    assert "proportions_pred" in adata.obsm
+    assert "proportions_true" in adata.obsm
+    return True
+
+
+def sample_dataset():
```

Comment:
`check_dataset(sample_dataset())` will fail as this does not match your defined API.